In [8]:
import pandas as pd

# Membaca dataset
path = "C:\\Users\\MSII\\Downloads\\GlobalWeatherRepository.csv"
df = pd.read_csv(path)

# Melihat korelasi dari tiga variabel yang diminta
df[['temperature_fahrenheit',
   'pressure_mb',
   'air_quality_Sulphur_dioxide']].corr()

,temperature_fahrenheit,pressure_mb,air_quality_Sulphur_dioxide
temperature_fahrenheit,1.000000,-0.486833,-0.023152
pressure_mb,-0.486833,1.000000,0.018620
air_quality_Sulphur_dioxide,-0.023152,0.018620,1.000000


# 1. Analisis korelasi
 Korelasi positif (+) → jika variabel pertama meningkat, variabel kedua cenderung meningkat.
Korelasi negatif (-) → jika variabel pertama meningkat, variabel kedua cenderung menurun.
Mendekati 0 → hubungan linear lemah.
Mendekati +1 atau -1 → hubungan linear kuat.

In [11]:
import pandas as pd

path = "C:\\Users\\MSII\\Downloads\\shopping_trends.csv"

df = pd.read_csv(path)

previous_purchases = df["Previous Purchases"]

print("=== Statistik Previous Purchases ===")
print("Mean :", previous_purchases.mean())
print("Std  :", previous_purchases.std())
print("Min  :", previous_purchases.min())
print("Q1   :", previous_purchases.quantile(0.25))
print("Q2   :", previous_purchases.quantile(0.50))
print("Q3   :", previous_purchases.quantile(0.75))
print("Max  :", previous_purchases.max())

=== Statistik Previous Purchases ===
Mean : 25.35153846153846
Std  : 14.447125170462309
Min  : 1
Q1   : 13.0
Q2   : 25.0
Q3   : 38.0
Max  : 50


In [12]:
df[['Age', 'Purchase Amount (USD)']].corr()

,Age,Purchase Amount (USD)
Age,1.000000,-0.010424
Purchase Amount (USD),-0.010424,1.000000


# 1. Korelasi Age vs Purchase Amount (USD)
Age vs Purchase Amount (USD) = -0.010424

Analisis:

Age vs Purchase Amount (USD) (-0.010424), berarti terdapat hubungan negatif yang sangat lemah antara usia pelanggan dengan jumlah pembelian. Nilai korelasi yang sangat mendekati 0 menunjukkan bahwa perubahan usia pelanggan hampir tidak memiliki hubungan dengan jumlah Purchase Amount (USD).

Kesimpulan:

Semakin bertambah usia pelanggan, jumlah pembelian tidak mengalami perubahan yang berarti. Dengan demikian, Age dan Purchase Amount (USD) hampir tidak memiliki hubungan linear.

In [13]:
df[['Purchase Amount (USD)', 'Review Rating']].corr()

,Purchase Amount (USD),Review Rating
Purchase Amount (USD),1.000000,0.030776
Review Rating,0.030776,1.000000


# 2. Korelasi Purchase Amount (USD) vs Review Rating
Hasil korelasi:

Purchase Amount (USD) vs Review Rating = +0.030776

Analisis:

Purchase Amount (USD) vs Review Rating (+0.030776), berarti terdapat hubungan positif yang sangat lemah antara jumlah pembelian dengan rating ulasan. Artinya, ketika jumlah pembelian meningkat, rating cenderung meningkat sedikit, tetapi hubungan tersebut sangat lemah karena nilai korelasinya mendekati 0.

Kesimpulan:

Besarnya Purchase Amount (USD) hampir tidak memiliki hubungan linear dengan Review Rating. Jadi, pelanggan yang melakukan pembelian dengan jumlah lebih besar belum tentu memberikan rating yang lebih tinggi.